### ISSUE TITLE
Accessibility bug: website home page color contrast issues in non-selected tabs

### ISSUE THREAD

**Issue by ashleemboyer** [tailwindlabs/headlessui#2826]
On [the Headless UI home page](https://headlessui.com/), the non-selected tab in the React & Vue tab set does not meet color contrast standards set in [WCAG 2.1 SC 1.4.3 Contrast (Minimum) (Level AA)](https://www.w3.org/TR/WCAG21/#contrast-minimum).

Here is a screenshot showing the issue where the Vue tab's text color has insufficient contrast with the background. 

<img width="190" alt="" src="https://github.com/tailwindlabs/headlessui/assets/43934258/129b36f3-0a26-4dfe-bbc4-b768c83dee1e">

**Comment by reinink** [comment]
Hey thanks for sharing this feedback.

We try hard to be mindful of color contrast guidelines in everything that we design, but there will definitely be some places that don't pass where we've made the conscious trade-off to prioritize the visual design over contrast guidelines. This situation is one example of that.

I appreciate that for some people this is an unacceptable decision — to that all I can say is that we do try very, very hard to be in the 99th percentile of the internet from an accessibility perspective, and I hope we get credit for the effort we put in that almost no other websites match.

Color contrast is the only place where we ever deliberately make any exceptions, because sometimes we need to balance strict guideline adherence with producing visual designs that can compete with the aesthetic of companies like Stripe or Linear, which is what a huge number of our customers are looking for from us.

That all said, for these particular tabs on the Headless UI home page, I have made a small tweak to increase the contrast of this text when the user has "increased contrast" enabled on their system — that should help out a little.

Hope that's all understandable 👍

**Comment by reinink** [comment]
Hey! Just a quick follow up on this. I was looking into the contrast level for this text using the Chrome developer tools, and apparently this text does actually pass the minimum contrast requirements:

<img width="518" alt="Screen Shot 2023-11-10 at 4 07 17 PM" src="https://github.com/tailwindlabs/headlessui/assets/882133/e3bdce59-656b-4f5a-a43b-7dc2605fb78a">

This is because the text-size is quite large (20px), which passes the contrast guidelines, as stated:

> [Large-scale](https://www.w3.org/TR/WCAG21/#dfn-large-scale) text and images of large-scale text have a contrast ratio of at least 3:1;

According to that document text needs to be at least 14px and bold, or 18px and regular. Since the links in our tabs are 20px and bold, they are more than large enough to pass this contrast requirements.

Hope that extra information helps 👍

**Comment by ashleemboyer** [comment]
So I found this issue using the [axe browser extension](https://www.deque.com/axe/devtools): [Elements must meet minimum color contrast ratio thresholds](https://dequeuniversity.com/rules/axe/4.8/color-contrast). I believe it's detecting the issue because `font-weight: 600` isn't bold, but semi-bold.

To clarify "large-scale" for posterity -- it names `pt` units, not `px`. From [Understanding SC 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#intent) (the first Note in the Intent section):

> When evaluating this Success Criterion, the font size in points should be obtained from the user agent or calculated on font metrics in the way that user agents do. Point sizes are based on the CSS pt size as defined in [CSS3 Values](https://www.w3.org/TR/css-values-3/#reference-pixel). The ratio between sizes in points and CSS pixels is 1pt = 1.333px, therefore 14pt and 18pt are equivalent to approximately 18.5px and 24px.

The text size is good since it's > `18.5px`, and the font weight is close but isn't quite there.

---

To respond to the first comment: It's unfortunate that visual marketing is prioritized over accessibility. Being accessible is an important to a significant portion of the population. They'll find someone else once they discover they're not a priority.

> Hey thanks for sharing this feedback.
>
> We try hard to be mindful of color contrast guidelines in everything that we design, but there will definitely be some places that don't pass where we've made the conscious trade-off to prioritize the visual design over contrast guidelines. This situation is one example of that.
>
> I appreciate that for some people this is an unacceptable decision — to that all I can say is that we do try very, very hard to be in the 99th percentile of the internet from an accessibility perspective, and I hope we get credit for the effort we put in that almost no other websites match.
>
> Color contrast is the only place where we ever deliberately make any exceptions, because sometimes we need to balance strict guideline adherence with producing visual designs that can compete with the aesthetic of companies like Stripe or Linear, which is what a huge number of our customers are looking for from us.
>
> That all said, for these particular tabs on the Headless UI home page, I have made a small tweak to increase the contrast of this text when the user has "increased contrast" enabled on their system — that should help out a little.
>
> Hope that's all understandable 👍
